pkg/components/util/helm.go

```diff
@@ -207,3 +212,88 @@ func chartFromManifests(metadata components.Metadata, manifests map[string]strin
 
 	return ch, nil
 }
+
+// ObjectMetadata uniquely identifies any object in the list of YAML manifests.
+type ObjectMetadata struct {
+	Version string
+	Kind    string
+	Name    string
+}
+
+// YAMLToUnstructured accepts a Kubernetes manifest in YAML format and returns an object of type
+// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
+// extract metadata from the Kubernetes manifest.
+func YAMLToUnstructured(yamlObj []byte) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+
+	// Decode YAML into `unstructured.Unstructured`.
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode(yamlObj, nil, u); err != nil {
+		return nil, fmt.Errorf("converting config to unstructured.Unstructured: %w", err)
+	}
+
+	return u, nil
+}
+
+// removeYAMLComments converts YAML to JSON and back again, this removes the comments in the YAML
+// and any extra whitespaces spaces.
+func removeYAMLComments(yamlObj []byte) ([]byte, error) {
+	jsonObj, err := yaml.YAMLToJSON(yamlObj)
+	if err != nil {
+		return nil, fmt.Errorf("converting YAML to JSON: %w", err)
+	}
+
+	yamlObj, err = yaml.JSONToYAML(jsonObj)
+	if err != nil {
+		return nil, fmt.Errorf("converting JSON to YAML: %w", err)
+	}
+
+	return yamlObj, nil
+}
+
+// SplitYAMLDocuments converts a YAML string with multiple YAML docs separated by `---` into unique
+// objects and returns those objects as a map.
+func SplitYAMLDocuments(yamlObj string) (map[ObjectMetadata]string, error) {
```

Comment:
To reflect our private conversation, I mistakenly pointed you @surajssd to `pkg/components/util` package, when I was actually thinking about `pkg/k8sutil.LoadManifests`, where we already have logic for parsing Kubernetes YAMLs to get their Kind, APIVersion etc. 

I think we should re-use this code in this tests. if you pass `yamlObj` to `pkg/k8sutil.LoadManifests`, then your helper may use `Kind()`, `Name()` and (non existing) `Namespace()` to find you the right object requested by test definition.

Reply:
Moved the code to `k8sutil`, going forward we can start using this API when getting rid of the manifests API. The API has issues like the struct itself is private and has functions that are public. This causes issues like we cannot pass the objects around in the functions. So in the refactor we can start consuming that.